lamagraph-compiler/src/Lamagraph/Compiler/ModuleResolver/Helper.hs

```diff
@@ -0,0 +1,51 @@
+module Lamagraph.Compiler.ModuleResolver.Helper (prependModuleName, lookupName, resolveMany) where
+
+import Relude
+
+import Control.Lens
+import Data.HashMap.Strict qualified as HashMap
+import Data.HashSet ()
+import Data.List.NonEmpty as NonEmpty
+
+import Lamagraph.Compiler.ModuleResolver.MrTypes
+import Lamagraph.Compiler.Syntax.Longident
+
+prependModuleName :: ModulePath -> Text -> FullName
+prependModuleName path name = FullName $ Longident $ coerce path <> NonEmpty.singleton name
+
+lookupInModule :: ModuleRegistry -> ModulePath -> Text -> Maybe FullName
+lookupInModule (ModuleRegistry reg) path name =
+  HashMap.lookup path reg >>= find (== name) <&> prependModuleName path
```

Comment:
It a bad practice to use `find` on (unordered-)containers, because it'll call `toList` first. As I see here, you just need to check whether `name` is really present in Module. Maybe it's easier to use `member` from `HashSet`.

Reply:
Also, can this be written using `case` or `do` to help with readability?